components/dialog/dialog.js

```diff
@@ -228,6 +228,12 @@ class Dialog extends PropertyRequiredMixin(LocalizeCoreElement(AsyncContainerMix
 		this.resize();
 	}
 
+	_handleSlotChange() {
+		requestAnimationFrame(() => {
+			this._checkForFocusableContentElem();
```

Comment:
This fixes things if there's a slotchange, but if there's not (e.g., display:none -> display: something content) it's not fixed for that (though that feels like potentially an abnormal case).